onnxruntime/core/providers/vitisai/vitisai_execution_provider.cc

```diff
@@ -76,4 +296,62 @@ common::Status VitisAIExecutionProvider::Compile(const std::vector<FusedNodeAndG
   }
   return Status::OK();
 }
+
+std::string VitisAIExecutionProvider::GetBackendCompileCacheDir() const {
+  if (info_.count("cacheDir") > 0) {
+    const std::string& cache_dir = info_.at("cacheDir");
+    if (!cache_dir.empty()) {
+      return cache_dir;
+    }
+  }
+  std::string cache_dir = ParseEnvironmentVariableWithDefault<std::string>("XLNX_CACHE_DIR", "");
+  if (!cache_dir.empty()) {
+    return cache_dir;
+  }
+  auto user_name = ParseEnvironmentVariableWithDefault<std::string>(
+      "USERNAME", ParseEnvironmentVariableWithDefault<std::string>("USER", ""));
+  std::string temp_dir =
+#ifdef _WIN32
+      "C:/temp/";
```

Comment:
how to guarantee the write access to this temp folder?

Reply:
Not guaranteed. It relies on `%TEMP%` to be writeable on Windows, which is true for most cases, otherwise many apps won't be able to function correctly.